pkg/cri/server/runtime_config.go

```diff
@@ -0,0 +1,83 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"sort"
+
+	"github.com/containerd/containerd/log"
+	runcoptions "github.com/containerd/containerd/runtime/v2/runc/options"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+// RuntimeConfig returns configuration information of the runtime.
+func (c *criService) RuntimeConfig(ctx context.Context, r *runtime.RuntimeConfigRequest) (*runtime.RuntimeConfigResponse, error) {
+	resp := &runtime.RuntimeConfigResponse{
+		Linux: &runtime.LinuxRuntimeConfiguration{
+			CgroupDriver: c.getCgroupDriver(ctx),
+		},
+	}
+	return resp, nil
+}
+
+func (c *criService) getCgroupDriver(ctx context.Context) runtime.CgroupDriver {
+	// Go through the runtime handlers in a predictable order, starting from the
+	// default handler, others sorted in alphabetical order
```

Comment:
let's have a conversation about defaulting... see if there isn't a better way than alphabetical, and / or see if there is an issue here:

Noting for the conversation that we probably need to consider adding validation against our config, and talk about other considerations if a runtime is requested that is not the same as reported via RuntimeConfig.

Reply:
> let's have a conversation about defaulting... see if there isn't a better way than alphabetical, and / or see if there is an issue here:

Let's have a chat.
 
> Noting for the conversation that we probably need to consider adding validation against our config, and talk about other considerations if a runtime is requested that is not the same (different cgroup driver) than as reported via RuntimeConfig.

Sounds pretty much like converging back to the old faithful "global" `systemd_cgroup` setting albeit in a very complicated fashion

Reply:
that might be the correct path until we can report said global and per/runtime  ... and the validation would let admins know..